src/python/pants/base/payload_field.py

```diff
@@ -139,8 +139,12 @@ def _compute_fingerprint(self):
     hasher.update(self._rel_path)
     for source in sorted(self.relative_to_buildroot()):
       hasher.update(source)
-      with open(os.path.join(get_buildroot(), source), 'rb') as f:
-        hasher.update(f.read())
+      abs_path = os.path.join(get_buildroot(), source)
+      if not os.path.exists(abs_path):
+        hasher.update('\a\n\n\t[FILE DOES NOT EXIST]\t\n\n\a')
+      else:
```

Comment:
Perhaps I should either add a virtual_sources=[...] argument to wire_gen to allow specifying sources that don't actually exist (changing the accompanying pom_to_build to generate that automatically), or change the java_wire_library to automatically split apart sources based on whether the exist.

I've prototyped both options, and they both work. The first option is probably better in terms of the robustness of the pants codebase, but means that my change to our pom_to_build script would break pants in our java repo if pants isn't updated with it (because wire will get a virtual_sources argument it doesn't understand).